### ISSUE TITLE
Omitting default actions if already provided

### ISSUE THREAD

**Issue by fabiospampinato** [sindresorhus/electron-context-menu#70]
I'm defining a custom "Copy" action, which does something a bit different than the default "Copy" action, to some buttons, but on right click the content of the button gets selected (and I can't make them unselectable) and so the menu displays 2 "Copy" actions, the one I defined and the one this library defines:

<img width="233" alt="Screen Shot 2019-05-09 at 18 14 34" src="https://user-images.githubusercontent.com/1812093/57469173-517f6b80-7286-11e9-934a-68e6b70e949c.png">

I think I could generate the whole menu manually via the `menu` option but that would mean detecting all the cases where the default actions should be showed.

Could we instead add an option, `canOverrideDefaultActions` or something, where before inserting each default action the library check if another action with the same label has been provided already?

**Comment by sindresorhus** [comment]
Makes sense, but should be based on `id`, not the label. I don't think we need an option for it. Should the replaced item take the original position or be added to the start/end depending on whether prepend/append was used? I'm leaning towards the latter.

We currently have a `shouldShowMenu` option, so it could also maybe be useful with one for menu items called `shouldShowMenuItem`? In case you want to dynamically decide which of the items to show.

**Comment by fabiospampinato** [comment]
> Makes sense, but should be based on id, not the label.

Maybe if no id is explicitly provided but the label is the same as a default menu item that default menu item should still be omitted? I just can't imagine a scenario where one would want 2 "Copy" menu items like shown in the screenshot.

> Should the replaced item take the original position or be added to the start/end depending on whether prepend/append was used?

I'd say the latter too.

> We currently have a shouldShowMenu option, so it could also maybe be useful with one for menu items called shouldShowMenuItem? In case you want to dynamically decide which of the items to show.

Maybe, I'm currently updating the menus inside `shouldShowMenu` before showing them ([ref](https://github.com/fabiospampinato/notable-private/blob/987027f0bde0ec3d7a68c8593f59e26b37701b72/src/renderer/components/main/extra/context_menu.tsx#L60-L61)).

If there was a `shouldShowMenuItem` item option I think that could be useful in some scenarios, but I would personally not use it as whether or not some menu items should be shown may depend on some computation that I'd rather perform once for all items rather than for each time and I'd rather perform it next to the logic for updating the menu itself rather than in a different place.

Maybe an `updateMenu` option, maybe called with the _entire_ menu template before showing it, could be useful too.

I'd be happy to submit a PR implementing these features 👍

**Comment by sindresorhus** [comment]
> Maybe if no id is explicitly provided but the label is the same as a default menu item that default menu item should still be omitted? I just can't imagine a scenario where one would want 2 "Copy" menu items like shown in the screenshot.

The reason I want to base it on `id` is that menu items could be localized and that could create ambiguity. It's just easier to reason about if it's only based on the `id`.

> Maybe an updateMenu option, maybe called with the entire menu template before showing it, could be useful too.

:+1: That actually sounds more useful and flexible than my `shouldShowMenuItem` idea.

**Comment by sindresorhus** [comment]
> I see your point, but some apps don't deal with i18n at all, and those who do I guess would translate "Copy" from this library and the custom "Copy" item with the same string 🤔

I was referring to depending on overriding the "Copy" item here, then localizing and suddenly the override stops working as the labels no longer match.

**Comment by fabiospampinato** [comment]
Yeah yeah, but one would want to localize this library too and assuming both instances of "Copy" get localized to the same string (but languages are complicated, maybe this assumption can't be made) the override would still work.

**Comment by sindresorhus** [comment]
> but one would want to localize this library too

It would be weird to have to translate the menu item here when you don't use it and instead override it with your. These are many reasons why as stable ID is better.

**Comment by sindresorhus** [comment]
Fixed in `beaab0d`. A `prepend`/`append` item whose `id` matches a default action now replaces it, and keeps the position `prepend`/`append` gave it.

This also fixes the common case of `prepend: actions => [actions.copy({transform})]`, which used to show `Copy` twice.
